### PR TITLE
faustPhysicalModeling: init at 2.20.2

### PR DESCRIPTION
**Pull request by magnetophon**
###### Motivation for this change

This would have been one package if it wasn't for the different licences.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by magnetophon** [comment]
Oh yeah, this needs https://github.com/NixOS/nixpkgs/pull/82650 to build...

**Comment by magnetophon** [comment]
This is ready for merging, now  #82650 has been merged.

File: lib/licenses.nix [hunk 1]
```diff
@@ -649,6 +649,12 @@ lib.mapAttrs (n: v: v // { shortName = n; }) {
     url = http://metadata.ftp-master.debian.org/changelogs/main/d/debianutils/debianutils_4.8.1_copyright;
   };
 
+  stk = {
```
Comment (SuperSandro2000) [inline lib/licenses.nix:652]: Please put this into a separate commit with the message ``licenses: add stk``.

File: pkgs/applications/audio/faustPhysicalModeling/default.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  buildPhase = ''
+    cd examples/physicalModeling
+
+    for f in *MIDI.dsp;
+    do
+      faust2jaqt -time -vec  -midi -nvoices 8 -t 99999 $f
+    done
+
+    for f in *MIDI.dsp;
+    do
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/faustPhysicalModeling/default.nix:24]:
```suggestion
    for f in *MIDI.dsp; do
```

File: pkgs/applications/audio/faustPhysicalModeling/default.nix [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    sha256 = "1mm93ba26b7q69hvabzalg30dh8pl858nj4m2bb57pznnp09lq9a";
+  };
+
+  buildInputs = [ faust2jaqt faust2lv2 ];
+
+  buildPhase = ''
+    cd examples/physicalModeling
+
+    for f in *MIDI.dsp;
+    do
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/faustPhysicalModeling/default.nix:19]:
```suggestion
    for f in *MIDI.dsp; do
```

File: pkgs/applications/audio/faustPhysicalModeling/default.nix [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+      cp $f $out/bin/
+    done
+  '';
+
+  meta = {
+    description = "The physical models included with faust compiled as jack standalone and lv2 instruments";
+    homepage = https://github.com/grame-cncm/faust/tree/master-dev/examples/physicalModeling;
+    license = stdenv.lib.licenses.mit;
+    maintainers = [ stdenv.lib.maintainers.magnetophon ];
+  };
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/faustPhysicalModeling/default.nix:44]:
```suggestion
  meta = with stdenv.lib; {
    description = "The physical models included with faust compiled as jack standalone and lv2 instruments";
    homepage = https://github.com/grame-cncm/faust/tree/master-dev/examples/physicalModeling;
    license = licenses.mit;
    maintainers = with maintainers; [ magnetophon ];
  };
```

File: pkgs/applications/audio/faustStk/default.nix [hunk 5]
```diff
@@ -0,0 +1,45 @@
+{ stdenv, fetchFromGitHub, faust2jaqt, faust2lv2 }:
+stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/faustStk/default.nix:2]:
```suggestion
{ stdenv, fetchFromGitHub, faust2jaqt, faust2lv2 }:

stdenv.mkDerivation rec {
```

File: pkgs/applications/audio/faustStk/default.nix [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    sha256 = "1mm93ba26b7q69hvabzalg30dh8pl858nj4m2bb57pznnp09lq9a";
+  };
+
+  buildInputs = [ faust2jaqt faust2lv2 ];
+
+  buildPhase = ''
+    cd examples/physicalModeling/faust-stk
+
+    for f in *.dsp;
+    do
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/faustStk/default.nix:19]:
```suggestion
    for f in *.dsp; do
```

File: pkgs/applications/audio/faustStk/default.nix [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+      faust2jaqt -time -vec  -midi -nvoices 8 -t 99999 $f
+    done
+
+    for f in *.dsp;
+    do
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/faustStk/default.nix:24]:
```suggestion
    for f in *.dsp; do
```

File: pkgs/applications/audio/faustStk/default.nix [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+      faust2lv2  -time -vec -double -gui -nvoices 32 -t 99999 $f
+    done
+  '';
+
+  installPhase = ''
+    mkdir -p $out/lib/lv2
+    mv *.lv2/ $out/lib/lv2
+    mkdir -p $out/bin
+    for f in $(find . -executable -type f);
+    do
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/faustStk/default.nix:34]:
```suggestion
    for f in $(find . -executable -type f); do
```

File: pkgs/applications/audio/faustStk/default.nix [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+      cp $f $out/bin/
+    done
+  '';
+
+  meta = {
+    description = " included with faust compiled as jack standalone and lv2 instruments";
+    homepage = https://ccrma.stanford.edu/~rmichon/faustSTK/;
+    license = stdenv.lib.licenses.stk;
+    maintainers = [ stdenv.lib.maintainers.magnetophon ];
+  };
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/faustStk/default.nix:44]:
```suggestion
  meta = with stdenv.lib; {
    description = " included with faust compiled as jack standalone and lv2 instruments";
    homepage = https://ccrma.stanford.edu/~rmichon/faustSTK/;
    license = licenses.stk;
    maintainers = with maintainers; [ magnetophon ];
  };
```

File: pkgs/applications/audio/faustStk/default.nix [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+    for f in *.dsp;
+    do
+      faust2lv2  -time -vec -double -gui -nvoices 32 -t 99999 $f
+    done
+  '';
+
+  installPhase = ''
+    mkdir -p $out/lib/lv2
+    mv *.lv2/ $out/lib/lv2
+    mkdir -p $out/bin
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/faustStk/default.nix:32]:
```suggestion
    mkdir -p $out/lib/lv2 $out/bin
    mv *.lv2/ $out/lib/lv2
```

File: pkgs/applications/audio/faustPhysicalModeling/default.nix [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+      faust2lv2  -time -vec -double -gui -nvoices 32 -t 99999 $f
+    done
+  '';
+
+  installPhase = ''
+    mkdir -p $out/lib/lv2
+    mv *.lv2/ $out/lib/lv2
+    mkdir -p $out/bin
+    for f in $(find . -executable -type f);
+    do
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/faustPhysicalModeling/default.nix:34]:
```suggestion
    mkdir -p $out/lib/lv2 $out/bin
    mv *.lv2/ $out/lib/lv2
    for f in $(find . -executable -type f); do
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Also please add platforms to both meta sections and squash the commits together one for the license and one for each package.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 82642` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>faustPhysicalModeling</li>
    <li>faustStk</li>
  </ul>
</details>

File: lib/licenses.nix (hunk 1 above)
Comment (magnetophon) [reply lib/licenses.nix:706]: Thanks for the review.
All done.

**Comment by magnetophon** [comment]
@SuperSandro2000 Is this PR OK now?

File: pkgs/applications/audio/faustStk/default.nix [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+  installPhase = ''
+    mkdir -p $out/lib/lv2 $out/bin
+    mv *.lv2/ $out/lib/lv2
+    for f in $(find . -executable -type f); do
+      cp $f $out/bin/
+    done
+  '';
+  meta = with stdenv.lib; {
+    description = "the physical modeling instruments included with faust, compiled as jack standalone and lv2 instruments";
+    homepage = https://ccrma.stanford.edu/~rmichon/faustSTK/;
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/faustStk/default.nix:37]:
```suggestion
    homepage = "https://ccrma.stanford.edu/~rmichon/faustSTK/";
```

File: pkgs/applications/audio/faustStk/default.nix [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+
+  installPhase = ''
+    mkdir -p $out/lib/lv2 $out/bin
+    mv *.lv2/ $out/lib/lv2
+    for f in $(find . -executable -type f); do
+      cp $f $out/bin/
+    done
+  '';
+  meta = with stdenv.lib; {
+    description = "the physical modeling instruments included with faust, compiled as jack standalone and lv2 instruments";
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/faustStk/default.nix:36]:
```suggestion
    description = "The physical modeling instruments included with faust, compiled as jack standalone and lv2 instruments";
```

File: pkgs/applications/audio/faustStk/default.nix [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+  '';
+
+  installPhase = ''
+    mkdir -p $out/lib/lv2 $out/bin
+    mv *.lv2/ $out/lib/lv2
+    for f in $(find . -executable -type f); do
+      cp $f $out/bin/
+    done
+  '';
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/faustStk/default.nix:35]:
```suggestion
  meta = with lib; {
```

File: pkgs/applications/audio/faustStk/default.nix [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+
+  buildPhase = ''
+    cd examples/physicalModeling/faust-stk
+
+    for f in *.dsp; do
+      faust2jaqt -time -vec  -midi -nvoices 8 -t 99999 $f
+    done
+
+    for f in *.dsp; do
+      faust2lv2  -time -vec -double -gui -nvoices 32 -t 99999 $f
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/faustStk/default.nix:24]:
```suggestion
      faust2jaqt -time -vec  -midi -nvoices 8 -t 99999 $f
      faust2lv2  -time -vec -double -gui -nvoices 32 -t 99999 $f
```

File: pkgs/applications/audio/faustPhysicalModeling/default.nix [hunk 16]
```diff
@@ -0,0 +1,42 @@
...
+    mkdir -p $out/lib/lv2 $out/bin
+    mv *.lv2/ $out/lib/lv2
+    for f in $(find . -executable -type f); do
+      cp $f $out/bin/
+    done
+  '';
+
+  meta = with stdenv.lib; {
+    description = "The physical models included with faust compiled as jack standalone and lv2 instruments";
+    homepage = https://github.com/grame-cncm/faust/tree/master-dev/examples/physicalModeling;
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/faustPhysicalModeling/default.nix:37]:
```suggestion
  meta = with lib; {
    description = "The physical models included with faust compiled as jack standalone and lv2 instruments";
    homepage = "https://github.com/grame-cncm/faust/tree/master-dev/examples/physicalModeling";
```

File: pkgs/applications/audio/faustPhysicalModeling/default.nix [hunk 17]
```diff
@@ -0,0 +1,42 @@
...
+  buildInputs = [ faust2jaqt faust2lv2 ];
+
+  buildPhase = ''
+    cd examples/physicalModeling
+
+    for f in *MIDI.dsp; do
+      faust2jaqt -time -vec  -midi -nvoices 8 -t 99999 $f
+    done
+
+    for f in *MIDI.dsp; do
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/faustPhysicalModeling/default.nix:22]:
```suggestion
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
@ofborg eval
